### ISSUE TITLE
Cant post, webfinger seems not to be working

### ISSUE THREAD

**Issue by roelfrenkema** [nextcloud/social#1734]
**Describe the bug**
When I look for my account @admin@nextcloudto.roelfrenkema.com on an mastodon or friendica server I get a request but nothing happens. I am not found on the instances.

**To Reproduce**

Go to any mastodon or friendica server and look up @admin@nextcloudto.roelfrenkema.com

**Expected behavior**
That I am found in the search.

**Screenshots**

**Client details:**
 - Debian Buster on Raspberry Pi 4
 - Firefox
 - Version 102.10.0esr (64-bit)
 - Device: desktop

<details>
<summary>Server details</summary>

**Social app version:**

**Operating system**:

**Web server:**

**Database:**

**PHP version:**

**Nextcloud version:** (see Nextcloud admin page)

</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```

</details>

**Comment by roelfrenkema** [comment]
I did some reseach and there seems to be a problem with the webfinger. Compared one of Nextcloud top one to one of Mastodon . I currently suspect this to be the reason I cannot get any posts out either.

`
 1.
#  curl 'https://nextcloudto.roelfrenkema.com/.well-known/webfinger?resource=acct:admin@nextcloudto.roelfrenkema.com'
{"subject":"admin@nextcloudto.roelfrenkema.com","aliases":["https:\/\/nextcloudto.roelfrenkema.com\/apps\/social\/@admin","https:\/\/nextcloudto.roelfrenkema.com\/index.php\/u\/admin"],"links":[{"rel":"self","type":"application\/activity+json","href":"https:\/\/nextcloudto.roelfrenkema.com\/apps\/social\/@admin"},{"rel":"http:\/\/webfinger.net\/rel\/profile-page","type":"text\/html","href":"https:\/\/nextcloudto.roelfrenkema.com\/index.php\/u\/admin"},{"rel":"http:\/\/ostatus.org\/schema\/1.0\/subscribe","template":"https:\/\/nextcloudto.roelfrenkema.com\/index.php\/apps\/social\/ostatus\/follow\/?uri={uri}"}]}#             
 
2.
#  curl 'https://mastodon.social/.well-known/webfinger?resource=acct:yaahaa@mastodon.social'            
{"subject":"acct:yaahaa@mastodon.social","aliases":["https://mastodon.social/@yaahaa","https://mastodon.social/users/yaahaa"],"links":[{"rel":"http://webfinger.net/rel/profile-page","type":"text/html","href":"https://mastodon.social/@yaahaa"},{"rel":"self","type":"application/activity+json","href":"https://mastodon.social/users/yaahaa"},{"rel":"http://ostatus.org/schema/1.0/subscribe","template":"https://mastodon.social/authorize_interaction?uri={uri}"}]}#              
`

**Comment by srijansaxena11** [comment]
I still don't understand what is the issue. Posting is working fine for me. And so is web finger. Do you see any error in console? Also what version of app are you using?

**Comment by roelfrenkema** [comment]
To start with there is no error in the console. Obviously everything looks fine.

But if we have a look with curl we see that in the subject field the 'acct' part is missing.

Also I am not sure how the other side reacts to the escapes in the return JSON.

As said  a curl finger will reveal this. I believe the other side is silently dropping the request.

I am on RC1

**Comment by srijansaxena11** [comment]
webfinger without acct in response has no impact on posting anything. also it is weird that you are unable to post and there is no error. There should be something either in console or nextcloud logs. because i am able to post on same rc1.

**Comment by roelfrenkema** [comment]
I think I just had a little breakthrough with the webfinger. On friendica The url address ( being: https://nextcloudto.roelfrenkema.com/index.php/apps/social/@admin/ ) is recognised, not on mastodon. trying to connect the nextcloud account gives 2 errors

Het type netwerk kon niet gedetecteerd worden. Contact kan niet toegevoegd worden.
Het opgegeven profiel adres bevat geen adequate informatie.

Type of network could not be detected. Contact cannot be added. The profile address has no adequate info.

Er werden geen compatibele communicatieprotocols of feeds ontdekt.
Het @-stijl-identiteitsadres komt niet overeen met een nekend protocol of e-mailcontact.
Gebruik mailto: voor het adres om een e-mailcontrole af te dwingen.

No compatibel communication protocol or feeds detected
The @ stijl identity address is not a known protocol or email address.

**Comment by srijansaxena11** [comment]
Again I am confused. How is any of this related to posting anything at all? And what have Mastodon and/or Friendica got to do anything with posting on Nextcloud Social?

**Comment by Danie10** [comment]
@srijansaxena11 they all use the ActivityPub protocol, so all three can follow, reply, and have their posts seen across those networks. Webfinger seems to work from other networks, but I'm also waiting this thread as my NextCloud Social instance has the same error. Webfinger is broken, and although posts show locally, they are not being seen by my Mastodon account.

**Comment by roelfrenkema** [comment]
> Again I am confused. How is any of this related to posting anything at all? And what have Mastodon and/or Friendica got to do anything with posting on Nextcloud Social?

Am I on the right app? This is supposed to connect to the fediverse of which Friendica and Mastodon are a part. I am confused now.

**Comment by roelfrenkema** [comment]
> @srijansaxena11 they all use the ActivityPub protocol, so all three can follow, reply, and have their posts seen across those networks. Webfinger seems to work from other networks, but I'm also waiting this thread as my NextCloud Social instance has the same error. Webfinger is broken, and although posts show locally, they are not being seen by my Mastodon account.

Dani could you try and connect to my next loud instance from your nextcloud instance. @admin@nextcloudto.roelfrenkema.com

**Comment by Danie10** [comment]
Yes a post made on NextCloud Social should be seen, followed, liked from Mastodon or any other Fediverse network using ActivityPub. And on NextCloud Social an account there must likewise see Mastodon and other accounts. This was certainly how it was working when I started using it. Unless that has all changed, and it is now stand-alone without ActivityPub. But I've not seen any communication about such a change.

The whole point of using Social was that it was an ActivityPub node running under NextCloud.

**Comment by roelfrenkema** [comment]
Looks like we hooked up Andie no notification or request or anything. But at least it's a start. Did you get the message I sent you?

**Comment by Danie10** [comment]
@roelfrenkema I've tried a public mention post now for you, but I tried to my own Mastodon account 2 days ago and nothing seems to be replicating. I think Webfinger is needed for remote servers to recognise the account. So I'm waiting for some solution to try as I spent 2 whole days trying all sorts of different things and nothing worked. It used to work before so I don't want to keep it at, and I end up breaking my NextCloud instance. But Webfinger and NodeInfo are issues - possibly even related to the NextCloud code itself. Just that Social makes use of Webfinger whole most other NextCloud apps don't need it.

**Comment by roelfrenkema** [comment]
Nope nothing came in. And yes my idea too is the the webfinger is not properly formatted. I think the acct is necessary and the escapes should be removed from the response. The curl request can help debugging this. Strange though. I had hoped that as we are connected now I would have received you message. Incoming usually is no problem. So it all boils down to outgoing.

**Comment by srijansaxena11** [comment]
What if I add "acct:" to subject? Does anyone want to try out with this?

**Comment by roelfrenkema** [comment]
Sure I pull it as soon as your ready

**Comment by srijansaxena11** [comment]
How about #1740 ?
I created a Telegram group for now if people are interested: https://t.me/nextcloud_social

**Comment by roelfrenkema** [comment]
Not for me thanks. Maybe after the war.

**Comment by srijansaxena11** [comment]
I tried adding acct: to subject. It doesn't change anything. I still cannot find my social on mastodon and cannot follow from friendica.

**Comment by srijansaxena11** [comment]
Are you people able to follow local accounts on the same instance?

**Comment by srijansaxena11** [comment]
On friendica my social URL is somehow of localhost:
![image](https://user-images.githubusercontent.com/34964694/232334609-ba787811-440f-46bc-9f84-4cda0ecc0a79.png)
